From: "Liam R. Howlett" <Liam.Howlett@oracle.com>
Subject: mm: move unmap_vmas() declaration to internal header
Date: Mon, 12 Jun 2023 16:39:40 -0400

unmap_vmas() is not used outside of the core mm code.  Move the
definition to mm/internal.h

Link: https://lkml.kernel.org/r/20230612203953.2093911-4-Liam.Howlett@oracle.com
Signed-off-by: Liam R. Howlett <Liam.Howlett@oracle.com>
Cc: Peng Zhang <zhangpeng.00@bytedance.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 include/linux/mm.h |    4 ----
 mm/internal.h      |    4 ++++
 2 files changed, 4 insertions(+), 4 deletions(-)

--- a/include/linux/mm.h~mm-move-unmap_vmas-declaration-to-internal-header
+++ a/include/linux/mm.h
@@ -2282,10 +2282,6 @@ static inline void zap_vma_pages(struct
 	zap_page_range_single(vma, vma->vm_start,
 			      vma->vm_end - vma->vm_start, NULL);
 }
-void unmap_vmas(struct mmu_gather *tlb, struct maple_tree *mt,
-		struct vm_area_struct *start_vma, unsigned long start,
-		unsigned long end, bool mm_wr_locked);
-
 struct mmu_notifier_range;
 
 void free_pgd_range(struct mmu_gather *tlb, unsigned long addr,
--- a/mm/internal.h~mm-move-unmap_vmas-declaration-to-internal-header
+++ a/mm/internal.h
@@ -1095,6 +1095,10 @@ static inline int vma_iter_store_gfp(str
 	return 0;
 }
 
+void unmap_vmas(struct mmu_gather *tlb, struct maple_tree *mt,
+		struct vm_area_struct *vma, unsigned long start_addr,
+		unsigned long end_addr, bool mm_wr_locked);
+
 /*
  * VMA lock generalization
  */
_
